shell/packages/sandstorm-db/db.js

```diff
@@ -430,6 +432,29 @@ ApiTokens = new Mongo.Collection("apiTokens");
 //   ...
 // }
 
+ApiHosts = new Mongo.Collection("apiHosts");
+// Allows defining some limited static behavior for an API host when accessed unauthenticated. This
+// mainly exists to allow backwards-compatibility with client applications that expect to be able
+// to probe an API host without authentication to determine capabilities such as DAV protocols
+// supported, before authenticating to perform real requests. An app can specify these properties
+// when creating an offerTemplate.
+//
+// Each contains:
+//   _id:          apiHostIdHashForToken() of the corresponding API token.
+//   hash2:        hash(hash(token)), aka hash(ApiToken._id). Used to allow ApiHosts to be cleaned
```

Comment:
> Is there any benefit to this being hash(hash(token)) instead of just hash(token)?

I have this weird desire to make sure that it is impossible to map from an API hostname back to the ApiToken. It's possible I'm being silly.

> It's kinda weird that the static resources will be stored and served forever, even if the grain that caused them to be created has since been deleted.

I think deleting the grain also deletes all ApiTokens and therefore all ApiHosts?